Build failure: handbrake

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

Build *handbrake* on latest unstable https://github.com/NixOS/nixpkgs/commit/c5296fdd05cfa2c187990dd909864da9658df755

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/319716805

### Relevant log output

```console
error: Cannot build '/nix/store/nm4whxk6b2k65dan5q30l3x3dhf15qi0-ffmpeg-full-7.1.1.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/19vsyhcqpjknmkr4ybf9dxd9s32cdqa4-ffmpeg-full-7.1.1-lib
         /nix/store/26n6fwkvrkky7y8z41c5zfi7lw3yhq0n-ffmpeg-full-7.1.1-bin
         /nix/store/7m77lkx839q7v6lg4zjjq8bsvch0fxx3-ffmpeg-full-7.1.1-dev
         /nix/store/ckpi27c03v8xaczn8ba0sdx95qb06lbk-ffmpeg-full-7.1.1-data
         /nix/store/csx21rshzgpf1g2qgmkamsylw8a7gz41-ffmpeg-full-7.1.1-man
         /nix/store/f76phrx0x72z87g2bnhrfjk9kvmz1jd5-ffmpeg-full-7.1.1-doc
         /nix/store/x23a7fa70xgwx6vsdql8my6cwib8d104-ffmpeg-full-7.1.1
       Last 25 log lines:
       > CC	libavcodec/qdmc.o
       > CC	libavcodec/qdrw.o
       > CC	libavcodec/qoadec.o
       > In file included from libavcodec/tableprint_vlc.h:40,
       >                  from libavcodec/qdm2_tablegen.c:24:
       > libavcodec/vlc.c: In function 'ff_vlc_init_multi_from_lengths':
       > libavcodec/vlc.c:530:20: error: implicit declaration of function 'av_malloc'; did you mean 'av_mallocz'? [-Wimplicit-function-declaration]
       >   530 |     multi->table = av_malloc(sizeof(*multi->table) << nb_bits);
       >       |                    ^~~~~~~~~
       >       |                    av_mallocz
       > libavcodec/vlc.c:530:18: warning: assignment to 'VLC_MULTI_ELEM *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
       >   530 |     multi->table = av_malloc(sizeof(*multi->table) << nb_bits);
       >       |                  ^
       > In file included from libavcodec/qdm2_tablegen.c:26:
       > libavcodec/qdm2_tablegen.h: In function 'build_vlc':
       > libavcodec/qdm2_tablegen.h:120:30: warning: pointer targets in passing argument 4 of 'ff_vlc_init_from_lengths' differ in signedness [-Wpointer-sign]
       >   120 |                              &tab[0][1], 2, &tab[0][0], 2, 1,
       >       |                              ^~~~~~~~~~
       >       |                              |
       >       |                              const uint8_t * {aka const unsigned char *}
       > libavcodec/vlc.c:307:44: note: expected 'const int8_t *' {aka 'const signed char *'} but argument is of type 'const uint8_t *' {aka 'const unsigned char *'}
       >   307 |                              const int8_t *lens, int lens_wrap,
       >       |                              ~~~~~~~~~~~~~~^~~~
       > make: *** [ffbuild/common.mak:201: libavcodec/qdm2_tablegen.o] Error 1
       > make: *** Waiting for unfinished jobs....
       For full logs, run:
         nix log /nix/store/nm4whxk6b2k65dan5q30l3x3dhf15qi0-ffmpeg-full-7.1.1.drv
```

### Additional context

Related to https://github.com/NixOS/nixpkgs/pull/477464#issuecomment-3756420230

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.7, NixOS, 26.05 (Yarara), 26.05.20260123.c5296fd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.33.1+1`
 - nixpkgs: `/nix/store/8hx8jwa24q6rkzhca9iqy5ckk9rbphi1-source`

### Notify maintainers

@Anton-Latukha @wmertens 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/ffmpeg/generic.nix
+++ b/pkgs/development/libraries/ffmpeg/generic.nix
@@ -497,6 +497,12 @@ stdenv.mkDerivation (
           hash = "sha256-Ixkf1xzuDGk5t8J/apXKtghY0X9cfqSj/q987zrUuLQ=";
         })
       ]
+      ++ optionals (lib.versionAtLeast version "7.1.1" && lib.versionOlder version "7.1.3") [
+        (fetchpatch2 {
+          url = "https://git.ffmpeg.org/gitweb/ffmpeg.git/patch/d8ffec5bf9a2803f55cc0822a97b7815f24bee83";
+          hash = "sha256-lmSI5arShb2/W84FMnSNs3lb6rd5vWdUSzfU8oza0Ic=";
+        })
+      ]
       ++ optionals (lib.versionOlder version "7.1.2") [
         (fetchpatch2 {
           name = "unbreak-svt-av1-3.0.0.patch";

